### ISSUE TITLE
Application captures mouse cursor despite being in windowed mode or despite alt-tabbing

### ISSUE THREAD

**Issue by arkhanari** [tton-bugs-hunters/tton-bug-tracker#4]
OS: Manjaro Linux (KDE)
Application version: GOG.com
When alt-tabbing from the application the mouse cursor is still being captured by the game.

**Comment by shmerl** [comment]
As a workaround, set some shortcut for minimizing the window and use that instead of Alt+Tab. When minimized, the game releases the cursor.

**Comment by vv221** [comment]
Right, inconsistent behaviour is most probably a bug.
I still keep the "enhancement" tag as I think an option would be the best way to fix this (captive mouse can be wanted in some situations).

**Comment by tonurics** [comment]
This is a "feature" of the SDL Fullscreen Mode, you can disable it by running the game in Windowed Mode.

Switching the game to Windowed Mode in the display options, doesn't actually cause SDL to change Modes [even though the game appears in a Window]. You need to restart the game yourself for it to take affect.

I thought about making the default run option Fullscreen in Windowed Mode, but I was having some serious performance/stability issues with it. The issue is likely: Compositing in the Window Manager not playing nice with the game. Depending on the Window Manager it can be disabled in settings somewhere, but I feel this is beyond what a normal user would know to do. So for the best experience, for everyone: we default to SDL Fullscreen Mode.

---

The workarounds you guys have discovered are interesting. If I get some free time, I might look in to them further.

**Comment by shmerl** [comment]
So how some other Unity / SDL based games function with fullscreen for unity switched to On, and Alt+Tab working? Quite a number of them do it without problems.

**Comment by tonurics** [comment]
I can't really speak to how other people have Unity3D games running. It might be that they aren't using SDL, or they have come up with workarounds. But from what I've seen: a lot of games just run in Window Mode at Fullscreen (without borders).

I *think* SDL is sending that signal while in Window Mode and there may not be any performance impact  on your system. But the issue was with our official Linux support target: Ubuntu 16.04 [it uses UnityWM]. Making sure things work well out of the box on Ubuntu is top priority. And for whatever reason, it was not liking Window Mode on our test machines.

Ultimately, at the end of the day: this is a nice to have feature. But it's outside the scope of the gameplay experience. Most players will never try to switch tasks while in the middle of play. And for now, "advanced users" can simply switch to Window Mode [they just need to know that's an option].

P.S. I'm a Arch user myself, so I try to make a point to ensure: whatever work is done, works well for most distributions; even if Ubuntu is our hard target.

**Comment by shmerl** [comment]
@tonurics :
> Ultimately, at the end of the day: this is a nice to have feature. But it's outside the scope of the gameplay experience. Most players will never try to switch tasks while in the middle of play. And for now, "advanced users" can simply switch to Window Mode [they just need to know that's an option].

Yes, that's one option. But I think a better option would be to give a choice of fullscreen & window borderless or complete fullscreen. This way, the user won't need to fiddle with maximzing the windowed view. I.e. fullscreen itself can have a sub-option, which will be disabled by default (to address your presumably most common Ubuntu target), and advanced users can enable that extra mode if they prefer.
